src/main/java/com/google/impactdashboard/server/api_utilities/LogRetriever.java

```diff
@@ -0,0 +1,109 @@
+                .fromStream(new ByteArrayInputStream(System.getenv("SERVICE_ACCOUNT_KEY").getBytes()));
+          }
+        return credentials;
+        })
+        .build();
+    return new LogRetriever(LoggingClient.create(LoggingSettings.create(stub)));
+  }
+
+  private LogRetriever(LoggingClient logger) {
+    this.logger = logger;
+  }
+
+  /**
+   * Creates a {@code ListLogEntriesRequest} and retrieves all the relevant audit logs.
+   * @return A list of all the relevant audit log entries that are stored by the logging API
+   */
+  public Collection<LogEntry> listAuditLogs() {
+    String project_id = "projects/concord-intern"; // needs to be retrieved from resource manager
+    // May need tweaking once tested and
+    String filter = "resource.type = project AND severity = NOTICE AND " +
+        "protoPayload.methodName:SetIamPolicy";
+    //Test of ListLogEntriesRequest will be changed once logging is tested
+    ListLogEntriesRequest request = ListLogEntriesRequest.newBuilder().setFilter(filter)
+        .setOrderBy("timestamp desc").addResourceNames(project_id).build();
+    ListLogEntriesPagedResponse response = logger.listLogEntries(request);
+    return StreamSupport.stream(response.iterateAll().spliterator(), false)
+        .collect(Collectors.toList());
+    //Get resources one or multiple [PROJECT_ID], [ORGANIZATION_ID]
+    //[BILLING_ACCOUNT_ID], [FOLDER_ID]
+    //filter by audit log
+    //Create new ListLogEntriesRequest with the information
+    //Call for logs
+  }
+
+  /**
+   * Creates a {@code ListLogEntriesRequest} and retrieves all the relevant Recommendation logs.
+   * @return A list of all the relevant recommendation log entries that are stored by the logging API.
+   */
+  public Collection<LogEntry> listRecommendationLogs() {
+    String project_id = "projects/concord-intern"; // needs to be retrieved from resource manager
+    // May need tweaking once tested
+    String filter = "resource.type = recommender";
+    //Test of ListLogEntriesRequest will be changed once logging is tested
+    ListLogEntriesRequest request = ListLogEntriesRequest.newBuilder().setFilter(filter)
+        .setOrderBy("timestamp desc").addResourceNames(project_id).build();
+    ListLogEntriesPagedResponse response = logger.listLogEntries(request);
+    return StreamSupport.stream(response.iterateAll().spliterator(), false)
+        .collect(Collectors.toList());
+    //Get resources one or multiple [PROJECT_ID], [ORGANIZATION_ID]
+    //[BILLING_ACCOUNT_ID], [FOLDER_ID]
+    //filter by recommendation log
+    //Create new ListLogEntriesRequest with the information
+    //Call for logs
+  }
+
+  public static void main(String[] args) throws Exception {
+    LogRetriever logRetriever = LogRetriever.create();
+
+    IamBindingRetriever iamBindingRetriever = IamBindingRetriever.create();
+
+    iamBindingRetriever.listIAMBindingData(logRetriever.listAuditLogs(), "concord-intern",
+        "Concord Intern", "12345");
+  }
```

Comment:
can this be removed now that it presumably works?

Reply:
I just forgot to remove the main